Facade_Engine/Query/ExternalEdges.cs

```diff
@@ -0,0 +1,76 @@
+/*
+ * This file is part of the Buildings and Habitats object Model (BHoM)
+ * Copyright (c) 2015 - 2021, the respective contributors. All rights reserved.
+ *
+ * Each contributor holds copyright over their respective contributions.
+ * The project versioning (Git) records all such contribution source information.
+ *                                           
+ *                                                                              
+ * The BHoM is free software: you can redistribute it and/or modify         
+ * it under the terms of the GNU Lesser General Public License as published by  
+ * the Free Software Foundation, either version 3.0 of the License, or          
+ * (at your option) any later version.                                          
+ *                                                                              
+ * The BHoM is distributed in the hope that it will be useful,              
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of               
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                 
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using System;
+using BH.oM.Geometry;
+using BH.oM.Dimensional;
+using BH.Engine.Geometry;
+using BH.Engine.Spatial;
+using BH.oM.Facade.Elements;
+using BH.oM.Facade.SectionProperties;
+using BH.oM.Spatial.ShapeProfiles;
+using BH.oM.Analytical.Elements;
+using BH.oM.Physical.FramingProperties;
+using BH.oM.Reflection;
+using System.Collections.Generic;
+using BH.oM.Reflection.Attributes;
+using System.ComponentModel;
+using System.Linq;
+
+namespace BH.Engine.Facade
+{
+    public static partial class Query
+    {
+        /***************************************************/
+        /****              Public methods               ****/
+        /***************************************************/
+
+        [Description("Returns external edges from a collection of 2D elements.")]
+        [Input("elems", "Elements to get external edges from.")]
+        [Output("extEdges", "External edges of elements.")]
+        public static List<IElement1D> ExternalEdges(this IEnumerable<IElement2D> elems)
+        {
+            if (elems == null)
+                return null;
+
+            List<IElement1D> allEdges = new List<IElement1D>();
+            List<IElement1D> result = new List<IElement1D>();
+            foreach (IElement2D elem in elems)
+            {
+                allEdges.AddRange(elem.ExternalElementCurves());
+            }
```

Comment:
```suggestion
            List<IElement1D> allEdges = elems.SelectMany(x => x.ExternalElementCurves()).ToList();
            List<IElement1D> result = new List<IElement1D>();
```

Reply:
Left similar to previous to keep the casting logic of allEdges as List<IElement1D> since elems.SelectMany... returns List<ICurve>. ALl ears if theres abetter way to do that.